packages/server-core/src/user/user-avatar/user-avatar.hooks.ts

```diff
@@ -42,6 +45,20 @@ import {
   userAvatarResolver
 } from './user-avatar.resolvers'
 
+const ifNoEntry = async (context: HookContext<UserAvatarService>) => {
+  const data = (Array.isArray(context.data) ? context.data : [context.data]) as UserAvatarType[]
+
+  for (const item of data) {
+    const entry = await context.service.find({
+      userId: context.params.query.userId
+    })
+    if (entry.total === 0) {
+      await context.service.create({ userId: context.params.query.userId, avatarId: item.avatarId })
+    }
+  }
+  return
+}
```

Comment:
@DanielBelmes just a suggestion, this can be done in a single find and single create query instead of doing in a for loop. I think following block will always return same result:

```
const entry = await context.service.find({
      userId: context.params.query.userId
    })
```

And create call can access an array I guess.

Reply:
Gotcha. I thought so and was discussing with Josh and we just decided to make it look like the other hooks for consistency. But yea my original version was just the inside of the for loop and checking for data. The for loop iteration does act as a check for data which is neat.